service/internal/checksum/middleware_compute_input_checksum_test.go

```diff
@@ -254,94 +271,27 @@ func TestComputeInputPayloadChecksum(t *testing.T) {
 					"CRC32": "AAAAAA==",
 				},
 			},
-			"http no algorithm require checksum": {
-				optionsFn: func(o *computeInputPayloadChecksum) {
-					o.RequireChecksum = true
-				},
-				buildInput: middleware.BuildInput{
-					Request: func() *smithyhttp.Request {
-						r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
-						r.URL, _ = url.Parse("http://example.aws")
-						r.ContentLength = 11
-						r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
-						return r
-					}(),
-				},
-				expectHeader: http.Header{
-					"Content-Md5": []string{"XrY7u+Ae7tCTyyK7j1rNww=="},
-				},
-				expectContentLength: 11,
-				expectPayload:       []byte("hello world"),
-				expectChecksumMetadata: map[string]string{
-					"MD5": "XrY7u+Ae7tCTyyK7j1rNww==",
-				},
-			},
-			"http no algorithm require checksum header preset": {
-				optionsFn: func(o *computeInputPayloadChecksum) {
-					o.RequireChecksum = true
-				},
-				buildInput: middleware.BuildInput{
-					Request: func() *smithyhttp.Request {
-						r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
-						r.URL, _ = url.Parse("http://example.aws")
-						r.ContentLength = 11
-						r.Header.Set("Content-MD5", "XrY7u+Ae7tCTyyK7j1rNww==")
-						r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
-						return r
-					}(),
-				},
-				expectHeader: http.Header{
-					"Content-Md5": []string{"XrY7u+Ae7tCTyyK7j1rNww=="},
-				},
-				expectContentLength: 11,
-				expectPayload:       []byte("hello world"),
-				expectChecksumMetadata: map[string]string{
-					"MD5": "XrY7u+Ae7tCTyyK7j1rNww==",
-				},
-			},
-			"https no algorithm require checksum": {
-				optionsFn: func(o *computeInputPayloadChecksum) {
-					o.RequireChecksum = true
-				},
-				buildInput: middleware.BuildInput{
-					Request: func() *smithyhttp.Request {
-						r := smithyhttp.NewStackRequest().(*smithyhttp.Request)
-						r.URL, _ = url.Parse("https://example.aws")
-						r.ContentLength = 11
-						r = requestMust(r.SetStream(bytes.NewReader([]byte("hello world"))))
-						return r
-					}(),
-				},
-				expectHeader: http.Header{
-					"Content-Md5": []string{"XrY7u+Ae7tCTyyK7j1rNww=="},
-				},
-				expectContentLength: 11,
-				expectPayload:       []byte("hello world"),
-				expectChecksumMetadata: map[string]string{
-					"MD5": "XrY7u+Ae7tCTyyK7j1rNww==",
-				},
-			},
 			"http seekable": {
 				initContext: func(ctx context.Context) context.Context {
-					return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmCRC32))
+					return internalcontext.SetChecksumInputAlgorithm(ctx, string(AlgorithmCRC32C))
```

Comment:
It's fine to use 32C, but curious to hear why you changed this on this revision. It seems like existing values could have worked just as well

Reply:
just want to test different algorithm